core/trino-main/src/main/java/io/trino/memory/ClusterMemoryManager.java

```diff
@@ -116,10 +108,10 @@
     private final Map<String, RemoteNodeMemory> nodes = new HashMap<>();
 
     @GuardedBy("this")
-    private final Map<MemoryPoolId, List<Consumer<MemoryPoolInfo>>> changeListeners = new HashMap<>();
+    private final List<Consumer<MemoryPoolInfo>> changeListeners = new ArrayList<>();
 
     @GuardedBy("this")
-    private final Map<MemoryPoolId, ClusterMemoryPool> pools;
+    ClusterMemoryPool pool;
```

Comment:
`private final`

Reply:
does it still need to be `@GuardedBy("this")`?

Reply:
dropped @Guarded (CMP is thread safe)